### PR TITLE
Handle errors thrown within query action clause

### PR DESCRIPTION
**Pull request by niveathika**
## Purpose
$subject

Related to https://github.com/ballerina-platform/ballerina-standard-library/issues/3611

## Checklist
- [ ] Linked to an issue
- [ ] Updated the specification
- [ ] Updated the changelog
- [ ] Added tests

### CODE REVIEW

File: ballerina/sql_client.bal [hunk 1]
```diff
@@ -207,7 +207,12 @@ public client class SQLClient {
                 record {}[] arr = [];
                 error? e = from record {} item in joinStream
                     do {
-                        arr.push(check item.cloneWithType(joinMetadata.entity));
+                        record {}|error clonedItem = item.cloneWithType(joinMetadata.entity);
+                        if clonedItem is record {} {
+                            arr.push(clonedItem);
+                        } else {
+                            return <Error>clonedItem;
+                        }
```
Comment (pcnfernando) [inline ballerina/sql_client.bal:215]: Can we add tests for the error case?

Comment (niveathika) [reply ballerina/sql_client.bal:215]: Is there a specific need? Technically the type for the table would not be changed dueing code execution.

Comment (pcnfernando) [reply ballerina/sql_client.bal:215]: Why do we need to do `cloneWithType` then?

Comment (niveathika) [reply ballerina/sql_client.bal:215]: Looping in @kaneeldias

Comment (kaneeldias) [reply ballerina/sql_client.bal:215]: `cloneWithType()` is not necessary. Will send a PR to fix

**Comment by niveathika** [comment]
Closing this in favour of https://github.com/ballerina-platform/module-ballerina-persist/pull/63#pullrequestreview-1182537945

**Pull request closed without merging**